### ISSUE TITLE
[Bug] Whisper Web does not work in incognito

### ISSUE THREAD

**Issue by jozefchutka** [xenova/transformers.js#162]
When running https://huggingface.co/spaces/Xenova/whisper-web in chrome incognito mode, it fails to transcribe and throws console errors:

```
whisper-web:195 Error while parsing the 'sandbox' attribute: 'allow-storage-access-by-user-activation' is an invalid sandbox flag.
script.js:1     POST https://huggingface.co/api/event 403
t @ script.js:1
p @ script.js:1
(anonymous) @ script.js:1
(anonymous) @ script.js:1
worker-73961048.js:1790 Uncaught (in promise) DOMException: Failed to execute 'open' on 'CacheStorage': An attempt was made to break through the security policy of the user agent.
```

Please consider if this is something worth to be fixed.

**Comment by PushpenderSaini0** [comment]
I believe this is caused as the website is loaded inside iFrame and the worker will have a different origin.
When trying to access caches, security policy will not allow this  as they have different origin !

There will not be any errors if you go directly to the website from [this](https://xenova-whisper-web.hf.space/index.html) URL.
This is the same link that is loaded in the iFrame.

**Comment by xenova** [comment]
Oh yes thanks for bringing this up! I ran into this problem a while ago and it's on my TODO list. I think the way to go about fixing this is to simply add a try-catch block around the `caches.open` call:

https://github.com/xenova/transformers.js/blob/15854f9cd6ec0bc08c67c4dec742d92d84b46092/src/utils/hub.js#L354

This will essentially disabled caching instead of crashing. Unless you know of another way to fix this?

**Comment by jozefchutka** [comment]
sounds good to me, worth to have a note in code though

**Comment by xenova** [comment]
Done 👍  https://github.com/xenova/transformers.js/commit/c491c2661fdd618542bd8a601ab48a761d459588

Just in time for v2.3.0 release too :)

**Comment by xenova** [comment]
These changes are now available in v2.3.0: https://www.npmjs.com/package/@xenova/transformers/v/2.3.0 🥳
